### ISSUE TITLE
Issue with date ranges in or statements

### ISSUE THREAD

**Issue by rknell** [balderdashy/sails-mongo#223]
Hello,

Just found one and I don't have time right now to debug this and do a pull request sorry.

When executing this code:

```javascript
CorporateInvoice.find({
        or: [
            {periodStart: {">=": periodStart, "=<": periodEnd}},
            {periodEnd: {">=": periodStart, "=<": periodEnd}}
        ],
        corporateAccountId: corporateAccountId
    })
```

Sails is generating this mongo statement:

```json
{
    "corporateAccountId": {},
    "$or": [
      {
        "periodStart": {
          "=<": {},
          "$gte": "2014-01-01"
        }
      },
      {
        "periodEnd": {
          "=<": {},
          "$gte": "2014-01-01"
        }
      }
    ]
  }
```
As you can see - the $lte isn't being parsed. I wrote the equivalent query in mongodb and it returns records:

```javascript
{corporateAccountId:"54a37464fd38764507bcd5cf", $or: [{periodStart:{$gte:"2014-01-01", $lte:"2015-01-01"}}, {periodEnd:{$gte:"2014-01-01", $lte: "2015-02-01"}}]}
```

**Comment by sailsbot** [comment]
Thanks for posting, @rknell.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
